src/signalrclient/callback_manager.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include<atomic>
+#include<unordered_map>
+#include<functional>
+#include<mutex>
+#include<cpprest\json.h>
+
+namespace signalr
+{
+    class callback_manager
+    {
+    public:
+        explicit callback_manager(const web::json::value& dtor_error);

```

Comment:
When in the dtor we need to clear all the callbacks to unblock the client who is awaiting on them. Since the callback takes a `const web::json::value&` we need to pass something. Since the `callback_manager` is just managing callbacks and not creating them it should not decide what to pass when a `callback_manager` instance is destroyed. Therefore it leaves it up to the consumer (which creates callbacks) to decide what to pass - since the consumer controls/creates the callbacks it may add logic to do something special in the callback if this value is passed to the callback.